compiler/src/iree/compiler/Dialect/LinalgExt/IR/test/canonicalize.mlir

```diff
@@ -35,3 +35,122 @@ func.func @sort_drop_unused_results(%arg0 : tensor<?x10xf32>,
 //  CHECK-SAME:     %[[ARG0:.+]]: tensor<?x10xf32>
 //  CHECK-SAME:     %[[ARG1:.+]]: tensor<?x10xi64>
 //       CHECK:   %[[SORT:.+]] = iree_linalg_ext.sort dimension(1) outs(%[[ARG0]] : tensor<?x10xf32>)
+
+// -----
+
+func.func @gather_to_extract_slice_expand(%source : tensor<1024x128xi32>, %indices : tensor<1x1xi32>) -> (tensor<1x1x128xi32>) {
+  %empty = tensor.empty() : tensor<1x1x128xi32>
+  %result = iree_linalg_ext.gather dimension_map = [0]
+    ins(%source, %indices : tensor<1024x128xi32>, tensor<1x1xi32>)
+    outs(%empty: tensor<1x1x128xi32>) -> tensor<1x1x128xi32>
+  return %result : tensor<1x1x128xi32>
+}
+
+// CHECK-LABEL: @gather_to_extract_slice_expand
```

Comment:
I don't know when do we start having a blank line in the between of the end of function and the start of checks. Can we remove them?

> The more code that fits on one screen, the easier it is to follow and understand the control flow of the program.

IMO, it falls into the cases that the blank line does not help readability. It's better if they can fit on one screen.

(There was an argument that the code and the checks should be interleaved, though I'm not convinced for some cases. Your style makes it slightly worse in this aspect.)

https://google.github.io/styleguide/cppguide.html#Vertical_Whitespace